local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php

```diff
@@ -57,25 +78,42 @@ public function generateUrlFunction($params, &$smarty)
             );
         }
 
+        if ($id && $router) {
+            // get url by router and id
+            /** @var Router $router */
+            $router = $this->container->get($router);
+            if (!$router) {
```

Comment:
This will never happen, if the service does not exists, an exception will be thrown (and I think its a `ServiceNotFoundException` )